### PR TITLE
Add Netty source itest #1036

### PR DESCRIPTION
**Pull request by tadayosi**
First part - camel-netty-kafka-connector source itest.

Will follow up with other itests later.

### CODE REVIEW

File: tests/itests-netty/src/test/java/org/apache/camel/kafkaconnector/netty/source/CamelNettyPropertyFactory.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public CamelNettyPropertyFactory withPort(int value) {
+        return setProperty("camel.source.path.port", value);
+    }
+
+    public CamelNettyPropertyFactory withSync(boolean value) {
+        return setProperty("camel.source.endpoint.sync", value);
+    }
+
+    public EndpointUrlBuilder<CamelNettyPropertyFactory> withUrl(String protocol, String host, int port, String options) {
+        String url = String.format("netty:%s://%s:%s%s", protocol, host, port, options == null ? "" : options);
```
Comment (orpiske) [inline tests/itests-netty/src/test/java/org/apache/camel/kafkaconnector/netty/source/CamelNettyPropertyFactory.java:45]: I think you don't need the `options` part. This method returns a `EndpointUrlBuilder` which allow us to compose the options (i.e..: `withUrl("udp", "localhost", 9090).append("option1", "value1").append("option2", "value2").buildUrl();`)

**Review by orpiske — CHANGES_REQUESTED** [review]
Thanks for the PR. I have just a single comment regarding the options handling on the `withUrl` method. Everything else looks good.

File: tests/itests-netty/src/test/java/org/apache/camel/kafkaconnector/netty/source/CamelNettyPropertyFactory.java (hunk 1 above)
Comment (tadayosi) [reply tests/itests-netty/src/test/java/org/apache/camel/kafkaconnector/netty/source/CamelNettyPropertyFactory.java:45]: Ah alright. Will change it and resubmit soon. Thanks!

**Review by orpiske — APPROVED** [review]
Looks great, thanks!

**Review by oscerd — APPROVED** [review]

**Pull request merged**